openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,110 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	mf "github.com/manifestival/manifestival"
+	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	EnableMonitoringEnvVar       = "ENABLE_SERVING_MONITORING_BY_DEFAULT"
+	EnableMonitoringLabel        = "openshift.io/cluster-monitoring"
+	ObservabilityCMName          = "observability"
+	ObservabilityBackendKey      = "metrics.backend-destination"
+	OpenshiftMonitoringNamespace = "openshift-monitoring"
+)
+
+const (
+	prometheusRoleName        = "knative-serving-prometheus-k8s"
+	prometheusClusterRoleName = "rbac-proxy-metrics-prom"
+)
+
+func ReconcileServingMonitoring(ctx context.Context, api kubernetes.Interface, ks *v1alpha1.KnativeServing) error {
+	backend, isSet := ks.Spec.CommonSpec.Config[ObservabilityCMName][ObservabilityBackendKey]
+	log := logging.FromContext(ctx)
+	if shouldEnableMonitoring(backend) {
+		log.Info("Enabling Serving monitoring")
+		if err := reconcileMonitoringLabelOnNamespace(ctx, ks.Namespace, api, true); err != nil {
+			return fmt.Errorf("failed to enable monitoring %w ", err)
+		}
+		return nil
+	}
+	log.Info("Disabling Serving monitoring")
+	if err := reconcileMonitoringLabelOnNamespace(ctx, ks.Namespace, api, false); err != nil {
+		return err
```

Comment:
nit:

```suggestion
		return fmt.Errorf("failed to disable monitoring %w ", err)
```

to be consisetnt with above.